bubble-wrap-style.yaml

```diff
@@ -2806,31 +2817,430 @@ layers:
-                                size: [[6,[16px,14px]],[13,[18px,15px]],[15,[22px,18px]]]
-                    width_3char:
-                        filter: function() { return (feature.shield_text.length === 3) }
+                                sprite: |
+                                    function() {
+                                        return ( 'BR:BR-' + feature.shield_text.length + 'char');
+                                    }
+                    france-a-and-n-road-ref-shields:
+                        filter: |
+                            function() {
+                                return ((feature.ref && ! feature.shield_text) &&
+                                        /^A .*$/.test(feature.ref) ||
+                                        /^N .*$/.test(feature.ref) ) ||
+                                    ((feature.ref && feature.network == 'e-road') &&
+                                    /^A .*$/.test(feature.ref) ||
+                                    /^N .*$/.test(feature.ref) )
+                            }
                         draw:
                             mapzen_icon_library:
-                                size: [[6,[20px,14px]],[13,[21px,15px]],[15,[26px,18px]]]
-                    width_4char:
-                        filter: function() { return (feature.shield_text.length === 4) }
+                                sprite: |
+                                    function() {
+                                        return ( 'FR:A-road-' + feature.ref.replace(" ", "").length + 'char');
+                                    }
+                                text:
+                                    text_source: |
+                                        function() {
+                                            return feature.ref.replace(" ", "");
+                                        }
+                    france-department-shields:
+                        filter: |
+                            function() {
+                                return ((feature.ref &&
+                                    ! feature.shield_text) &&
+                                    /^FR:.*:.*$/.test(feature.network) ||
+                                    /^D .*$/.test(feature.ref)) ||
+                                    ( feature.ref &&
+                                      /^D .*$/.test(feature.ref))
+                            }
                         draw:
                             mapzen_icon_library:
-                                size: [[6,[25px,14px]],[13,[28px,15px]],[15,[34px,18px]]]
-                    width_5char:
-                        filter: function() { return (feature.shield_text.length === 5) }
+                                sprite: |
+                                    function() {
+                                        var text_len = feature.shield_text.length ? feature.ref.length;
+                                        return ( 'FR:D-' + feature.text_len + 'char');
```

Comment:
Related to above, not sure this is what you actually mean here? Is there a `text_len` property on the feature itself? I assume you want to use the variable you just defined above, like:

`return ( 'FR:D-' + text_len + 'char');`

Reply:
fixed in https://github.com/tangrams/bubble-wrap/pull/262/commits/33743924702df949c53fee76bf59e38c4d1ad099